src/utils/vc.js

```diff
@@ -100,7 +93,66 @@ export function isRevocationCheckNeeded(credStatus, forceRevocationCheck, revoca
  */
 export function getDockRevIdFromCredential(credential) {
   // The hash outputs the same number of bytes as required by Dock
-  return blake2AsHex(credential.id, RevEntryByteSize * 8);
+  return blake2AsHex(credential['@id'], RevEntryByteSize * 8);
+}
+
+/**
+ * Helper method to ensure credential is valid according to the context
+ * @param credential
+ */
+export async function expandJSONLD(credential) {
+  const expanded = await jsonld.expand(credential);
+  return expanded[0];
+}
+
+/**
+ * Checks if a credential has a credentialStatus property and it has the properties we expect
+ * If status doesnt exist or is invalid, throws an error.
+ * @param expanded
+ */
+export async function getCredentialStatuses(expanded) {
+  const statusValues = jsonld.getValues(expanded, expandedStatusProperty);
+  if (statusValues.length === 0) {
+    return statusValues;
+  }
+
+  statusValues.forEach((status) => {
+    if (!status['@id']) {
+      throw new Error('"credentialStatus" must include an id.');
+    }
+    if (!status['@type']) {
+      throw new Error('"credentialStatus" must include a type.');
+    }
+  });
+
+  return statusValues;
+}
+
+/**
+ * Checks if a credential context is valid, ensures first context is https://www.w3.org/2018/credentials/v1
+ * @param credential
+ */
+export function checkCredentialContext(credential) {
+  if (credential['@context'][0] !== DEFAULT_CONTEXT) {
+    throw new Error(`${DEFAULT_CONTEXT} needs to be first in the list of contexts.`);
+  }
+}
+
+/**
+ * Check if a credential status has a Dock specific revocation
+ * @param status
+ * @returns {Boolean}
+ */
+function hasDockRevocation(status) {
+  const id = status['@id'];
```

Comment:
i disagree, id argue its cleaner to do

```
if (status
    && jsonld.getValues(status, credentialTypeField).includes(RevRegType)
    && id.startsWith(DockRevRegQualifier)
    && isHexWithGivenByteSize(id.slice(DockRevRegQualifier.length), RevRegIdByteSize)) {
    return true;
  }
```

instead of 

```
  if (status
    && jsonld.getValues(status, credentialTypeField).includes(RevRegType)
    && status[credentialIDField].startsWith(DockRevRegQualifier)
    && isHexWithGivenByteSize(status[credentialIDField].slice(DockRevRegQualifier.length), RevRegIdByteSize)) {
    return true;
  }
```